src/screens/TopicPageScreen/elements/NavHeader.js

```diff
@@ -197,17 +176,23 @@ const NavHeader = (props) => {
                   {isFollow === undefined && isLoading ? (
                     <Shimmer width={normalize(100)} height={normalize(36)} />
                   ) : isFollow ? (
-                    <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
+                    <ButtonFollowing
+                      handleSetUnFollow={onFollowButtonPress}
+                      followType={followType}
+                    />
                   ) : (
-                    <ButtonFollow handleSetFollow={handleFollowTopic} />
+                    <ButtonFollow handleSetFollow={onFollowButtonPress} />
                   )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleSetUnFollow` and `handleSetFollow` props have been updated to use `onFollowButtonPress` instead of `handleFollowTopic`. The `ButtonFollowing` component also now accepts a `followType` prop. These changes are in line with the earlier updates and the PR's purpose.
```diff
- <ButtonFollowing handleSetUnFollow={handleFollowTopic} />
- <ButtonFollow handleSetFollow={handleFollowTopic} />
+ <ButtonFollowing
+   handleSetUnFollow={onFollowButtonPress}
+   followType={followType}
+ />
+ <ButtonFollow handleSetFollow={onFollowButtonPress} />
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->